homeassistant/components/marytts/tts.py

```diff
@@ -49,57 +50,43 @@ class MaryTTSProvider(Provider):
     def __init__(self, hass, conf):
         """Init MaryTTS TTS service."""
         self.hass = hass
-        self._host = conf.get(CONF_HOST)
-        self._port = conf.get(CONF_PORT)
-        self._codec = conf.get(CONF_CODEC)
-        self._voice = conf.get(CONF_VOICE)
-        self._language = conf.get(CONF_LANG)
+        self._mary = MaryTTS(
+            conf.get(CONF_HOST),
+            conf.get(CONF_PORT),
+            conf.get(CONF_CODEC),
+            conf.get(CONF_LANG),
+            conf.get(CONF_VOICE),
+        )
+        self._effects = conf.get(CONF_EFFECT)
         self.name = "MaryTTS"
 
     @property
     def default_language(self):
         """Return the default language."""
-        return self._language
+        return self._mary.locale
 
     @property
     def supported_languages(self):
         """Return list of supported languages."""
         return SUPPORT_LANGUAGES
 
-    async def async_get_tts_audio(self, message, language, options=None):
-        """Load TTS from MaryTTS."""
-        websession = async_get_clientsession(self.hass)
-
-        actual_language = re.sub("-", "_", language)
-
-        try:
-            with async_timeout.timeout(10):
-                url = f"http://{self._host}:{self._port}/process?"
-
-                audio = self._codec.upper()
-                if audio == "WAV":
-                    audio = "WAVE"
+    @property
+    def default_options(self):
+        """Return dict include default options."""
+        return {CONF_EFFECT: self._effects}
 
-                url_param = {
-                    "INPUT_TEXT": message,
-                    "INPUT_TYPE": "TEXT",
-                    "AUDIO": audio,
-                    "VOICE": self._voice,
-                    "OUTPUT_TYPE": "AUDIO",
-                    "LOCALE": actual_language,
-                }
+    @property
+    def supported_options(self):
+        """Return a list of supported options."""
+        return SUPPORT_OPTIONS
 
-                request = await websession.get(url, params=url_param)
+    async def async_get_tts_audio(self, message, language, options=None):
+        """Load TTS from MaryTTS."""
+        effects = self._effects
 
-                if request.status != 200:
-                    _LOGGER.error(
-                        "Error %d on load url %s", request.status, request.url
-                    )
-                    return (None, None)
-                data = await request.read()
+        if options is not None and CONF_EFFECT in options:
+            effects.update(options[CONF_EFFECT])
```

Comment:
that means, you will be never able to overwrite the effects from config. Maybe:
`effects = options[CONF_EFFECT]`?

Reply:
I understood the options parameter as additional options during the service call.
Can you explain where the options parameter get its value?

I intended to use the config from yaml and let it be overwritten by service call options.

Reply:
options are they from the service call. I want only prevent to have a bug :)

Reply:
Since the file config is coming as default, it will change it to your suggestion.